packages/kbn-apm-config-loader/src/config.test.ts

```diff
@@ -41,6 +41,34 @@ describe('ApmConfiguration', () => {
     expect(config.getConfig('myservice').serviceVersion).toBe('9.2.1');
   });
 
+  it('allows overriding the service name via root config', () => {
+    const kibanaConfig = {
+      elastic: {
+        apm: {
+          serviceName: 'serviceNameOverride',
```

Comment:
I'm not an expert in APM but, is it OK to report all APM agents (NodeJS and RUM) under the same name?

How will the UI behave? AFAIK, it tends to show some features based on the type of agent.

Also, we initialize other services from this config: FTRs, Health Gateway, Dev Proxy, ...

Reply:
> Also, we initialize other services from this config: FTRs, Health Gateway, Dev Proxy, ...

It's not like we're planning to use this for our internally instrumented cluster, is it? So I'm not sure to understand the fear here. 

The real question is, if for any reasons a customer wants that, is there any problem letting him do it? And from my understanding, there's not, right?

Reply:
> As a reminder, we could already "name all our service with the same name" on main, by using per-service overrides with the same values.

Could we? AFAIK, we explicitly enforce the `serviceName`, no matter the overrides.

Reply:
> Having tried it, it doesn't "break the APM UI", it just reports everything under the same service name, which is absolutely and unarguably more confusing, but not breaking our app in any way.

It leads to unexpected, potentially breaking, behavior (see @vigneshshanmugam's comment [here](https://github.com/elastic/kibana/pull/181419#discussion_r1576734078)).

If we want to allow users to change it at their own risk, I'm ok to this change. I don't mean to oppose it. However, it could set false expectations without appropriate warnings in place. I wonder if there should be a clear recommendation of how to deal with this scenario (in the docs?), whether that's using the environment (pedantically, not correct because stacks/projects and environments are different concepts) or global labels.

That's why I'd rather confirm with the APM experts to make sure we understand the risks. Happy to approve anytime, though.